src/components/router/routerHelpers.js

```diff
@@ -29,6 +51,84 @@ const baseName = (helpers.TEST_MODE && '/') || (helpers.DEV_MODE && '/') || dyna
 const basePath =
   (helpers.TEST_MODE && '/') || (helpers.DEV_MODE && '/') || window.location.pathname.split(helpers.UI_NAME)[0];
 
+/**
+ * Generate product groups for applying query filter resets.
+ *
+ * @param {Array} config
+ * @returns {Array}
+ */
+const generateProductGroups = (config = routesConfig) => {
+  const productGroups = {};
+
+  config.forEach(({ pathParameter, productParameter }) => {
+    const viewIds = ((Array.isArray(productParameter) && productParameter) || [productParameter]).map(
+      id => (id && `view${id}`) || id
+    );
+
+    viewIds.forEach((id, index) => {
+      if (id) {
+        if (!productGroups[id]) {
+          productGroups[id] = [];
+        }
+
+        if (pathParameter) {
+          productGroups[id].push((Array.isArray(pathParameter) && pathParameter?.[index]) || pathParameter);
+        }
+      }
+    });
+  });
+
+  return productGroups;
+};
+
+/**
+ * Reference for products grouped by view.
+ */
+const productGroups = generateProductGroups();
+
+/**
+ * Generate routes to be consumed by router.
+ *
+ * @param {Array} config
+ * @returns {Array}
+ */
+const generateRoutes = (config = routesConfig) =>
+  config.map(
+    ({
+      activateOnError,
+      component,
+      disabled,
+      path: routePath,
+      pathParameter,
+      productParameter,
+      redirect,
+      ...routeConfig
+    }) => ({
+      activateOnError,
+      component,
+      disabled,
+      exact: true,
+      routeConfig: {
```

Comment:
optional: consider renaming this to "productConfig" its primary purpose is to provide identifiers to products

Reply:
slightly different direction